WooCommerce/Classes/ViewRelated/Products/View Models/ProductsTabProductViewModel.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+import Yosemite
+
+/// Converts the input product model to properties ready to be shown on `ProductsTabProductTableViewCell`.
+struct ProductsTabProductViewModel {
+    let imageUrl: String?
+    let name: String
+    let detailsAttributedString: NSAttributedString
+
+    init(product: Product) {
+        imageUrl = product.images.first?.src
+        name = product.name
+        detailsAttributedString = product.createDetailsAttributedString()
+    }
+}
+
+private extension Product {
+    func createDetailsAttributedString() -> NSAttributedString {
+        let statusText = createStatusText()
+        let stockText = createStockText()
+        let variationsText = createVariationsText()
+
+        let detailsText = [statusText, stockText, variationsText]
+            .compactMap({ $0 })
+            .joined(separator: " • ")
+
+        let attributedString = NSMutableAttributedString(string: detailsText,
+                                                         attributes: [
+                                                            .foregroundColor: StyleManager.wooGreyMid,
+                                                            .font: StyleManager.footerLabelFont
+            ])
+        if let statusText = statusText {
+            attributedString.addAttributes([.foregroundColor: StyleManager.highlightTextColor],
+                                           range: NSRange(location: 0, length: statusText.count))
+        }
+        return attributedString
+    }
+
+    func createStatusText() -> String? {
+        switch productStatus {
+        case .pending, .draft:
+            return productStatus.description
+        default:
+            return nil
+        }
+    }
+
+    func createStockText() -> String? {
+        switch productStockStatus {
+        case .inStock:
+            if let stockQuantity = stockQuantity {
+                let format = NSLocalizedString("%ld in stock", comment: "Label about product's inventory stock status shown on Products tab")
+                return String.localizedStringWithFormat(format, stockQuantity)
+            } else {
+                return NSLocalizedString("In stock", comment: "Label about product's inventory stock status shown on Products tab")
+            }
+        default:
+            return productStockStatus.description
+        }
+    }
+
+    func createVariationsText() -> String? {
+        guard !variations.isEmpty else {
```

Comment:
This is probably beyond the scope of this PR, but what about adding an `isNotEmpty`? That way this would read `guard variations.isNotEmpty` which I find much more readable

Reply:
🙇